app/controllers/transformation-controller.js

```diff
@@ -84,21 +84,22 @@ function getSurveyParts( req, res, next ) {
  * @param  {Function} next [description]
  * @return {[type]}        [description]
  */
-function getCachedSurveyHash( req, res, next ) {
+function getSurveyHash( req, res, next ) {
     _getSurveyParams( req.body )
         .then( function( survey ) {
             return cacheModel.getHashes( survey );
         } )
         .then( function( survey ) {
-            delete survey.credentials;
+            return _updateCache( survey );
+        } )
```

Comment:
I created a `then` section for `_updateCache`. I had to leave `return cacheModel.getHashes( survey );` which you had suggested we remove because for some reason I'm not getting a hash value from `_getCombinedHash` without it.